### PR TITLE
dependencies updated

### PR DESCRIPTION
**Pull request by cancerberoSgx**
Just updating dependencies (mostly typescript) solved an exception when generating using --includeDeclarations. Pasting the exception just in case:

```sh
node bin/typedoc  --mode modules --includeDeclarations --out tmp  --tsconfig ../cli-driver/packages/cli-driver/tsconfig.json ../cli-driver/packages/cli-driver/src  

Using TypeScript 2.7.2 from /home/sg/git/typedoc/node_modules/typescript/lib
/home/sg/git/typedoc/dist/lib/models/types/array.js:37
        var elementTypeStr = this.elementType.toString();
                                              ^

TypeError: Cannot read property 'toString' of undefined
    at ArrayType.toString (/home/sg/git/typedoc/dist/lib/models/types/array.js:37:47)
    at Object.escapeExpression (/home/sg/git/typedoc/node_modules/handlebars/dist/cjs/handlebars/utils.js:91:17)
    at eval (eval at createFunctionContext (/home/sg/git/typedoc/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23), <anonymous>:4:17)
    at Object.prog [as inverse] (/home/sg/git/typedoc/node_modules/handlebars/dist/cjs/handlebars/runtime.js:221:12)
    at ArrayType.<anonymous> (/home/sg/git/typedoc/node_modules/handlebars/dist/cjs/handlebars/helpers/if.js:17:22)
    at eval (eval at createFunctionContext (/home/sg/git/typedoc/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23), <anonymous>:5:35)
    at Object.prog [as inverse] (/home/sg/git/typedoc/node_modules/handlebars/dist/cjs/handlebars/runtime.js:221:12)
    at ArrayType.<anonymous> (/home/sg/git/typedoc/node_modules/handlebars/dist/cjs/handlebars/helpers/if.js:17:22)
    at eval (eval at createFunctionContext (/home/sg/git/typedoc/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23), <anonymous>:6:32)
    at Object.prog [as fn] (/home/sg/git/typedoc/node_modules/handlebars/dist/cjs/handlebars/runtime.js:221:12)

```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,9 +71,10 @@
     "LICENSE"
   ],
   "scripts": {
-    "test": "mocha -t 10000 dist/test",
+    "test": "mocha -t 10000 --exit dist/test",
```
Comment (aciccarello) [inline package.json]: Good catch. Didn't realize that got [changed in mocha 4](https://mochajs.org/#--exit----no-exit). The grunt mocha change may need to be changed too.

**Comment by ethanresnick** [comment]
Is there anything in particular holding this up?

**Comment by cancerberoSgx** [comment]
if you updated typescript that's fine for me, thanks!

**Comment by ryan-rowland** [comment]
@aciccarello If you're going to close other PRs for updating the TS version in favor of this PR, could you merge this one? This typescript version is having seems to be causing issues with many other dependencies. 2.8 and 2.9 work fine.

**Comment by huchim** [comment]
Typescript ^2.8 is required in @types/react-router (react-router 4)... 

Someone know if ETA exist for the next upgrade?

You may can edit the README to include a reference about incompatibility with Typescript ^2.8.

Thanks

**Comment by tyler-johnson** [comment]
I'd like to make a suggestion, however I know very little about this project and how typescript is integrated so please take it with a grain of salt.

What if, instead of shipping a specific version of TypeScript, TypeDoc used the version that was already installed in the user's project. This way the output of the documentation matches closer to the original types the user declared and users don't need to open an issue every time there is a minor version change to Typescript.

Now I imagine this wouldn't be easy to accomplish given that differences in API and features between versions. However other communities have managed to solve this through the use of version specific abstractions. Take Enzyme (React Testing Library) for example, they have separate "Adapters" for each React version to ensure compatibility.

**Comment by cancerberoSgx** [comment]
Is it really necessary to support node.js version 4 ? is quite old / deprecated so before investigating why is failing I'm asking. BTW I updated all dependencies, among then TypeScript to 2.9.2. Thanks

**Comment by leafoflegend** [comment]
The last release was 5 months ago. Can we please merge and release this PR? The delays on merging and releasing this is causing TypeDoc to become obsolete and seemingly development stagnant. Both material-ui and react-redux typings break this, and those are two very common and large libraries.

I cannot hold back using dependencies that are actually used in my production builds just to have auto-generated docs, but would like to use this. Can we either:

1. Take @tyler-johnson idea on this seriously (ideal).
2. Speed up the release process for this project. Id be fine even if it meant having a `beta` tag for the distribution so people can continue to use this library while you all finish your internal debate over how many things to fit into the next release.

**Pull request closed without merging**